api/main.py

```diff
@@ -44,11 +44,15 @@ def semantic():
 @app.route('/chat', methods=['POST'])
 @cross_origin()
 def chat():
-    query = request.json.get('query')
+    query = request.json.get('query', None)
     session_id = request.json.get('sessionId')
     history = request.json.get('history', [])
     settings = request.json.get('settings', {})
 
+    if query is None:
```

Comment:
this allows querying by:
* just a query string
* a query string + history
* just history (as the last item is going to be the query)